sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobApiTests.java

```diff
@@ -3072,15 +3072,15 @@ public void storageAccountAudience() {
         assertTrue(aadBlob.exists());
     }
 
+    @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "2024-08-04")
+    @LiveOnly
     @Test
-    public void audienceError() {
-        BlobClient aadBlob = instrument(new BlobClientBuilder().endpoint(bc.getBlobUrl())
-            .credential(new MockTokenCredential())
-            .audience(BlobAudience.createBlobServiceAccountAudience("badAudience")))
+    public void audienceErrorBearerChallengeRetry() {
+        BlobClient aadBlob = getBlobClientBuilderWithTokenCredential(bc.getBlobUrl())
+            .audience(BlobAudience.createBlobServiceAccountAudience("badAudience"))
```

Comment:
I'm intrigued by this change, does this mean bad audiences will now work?

Reply:
No, if this follows the implementation of .NET, we switch to the correct audience that was returned by the bearer challenge automatically and then request a new token.